src/benchnet.rs

```diff
@@ -0,0 +1,65 @@
+use crate::{input::*, model::*};
+use egg::*;
+
+fn resnet_block(
+    graph: &mut GraphConverter,
+    mut input: Id,
+    strides: (i32, i32),
+    out_channels: i32,
+    input_dim_1: i32,
+) -> Id {
+    let w1 = graph.new_weight(vec![out_channels, input_dim_1, 1, 1]);
+    let t = graph.conv2d(input, w1, 1, 1, PSAME, ACTRELU);
+    let w2 = graph.new_weight(vec![out_channels, out_channels, 3, 3]);
+    let t = graph.conv2d(t, w2, strides.0, strides.1, PSAME, ACTRELU);
+    let w3 = graph.new_weight(vec![out_channels * 4, out_channels, 1, 1]);
+    let t = graph.conv2d(t, w3, 1, 1, PSAME, ACTNONE);
+    if (strides.0 > 1) || (input_dim_1 != out_channels * 4) {
+        let w4 = graph.new_weight(vec![out_channels * 4, input_dim_1, 1, 1]);
+        input = graph.conv2d(input, w4, strides.0, strides.1, PSAME, ACTRELU);
+    }
+    let t = graph.add(input, t);
+    let t = graph.mul(input, t);
+    graph.relu(t)
+}
+
+/// Gets the RecExpr of a resnet50 model
+pub fn get_benchnet() -> RecExpr<Mdl> {
```

Comment:
I don't know benchnet, is it just resnet50 ?

Reply:
It's my hand-crafted network, it's resnet50 adding a few ops. It's the one I use to profile the more efficient implementation of two functions in egg: https://github.com/mwillsey/egg/pull/38